lib/paths-cache.js

```diff
@@ -346,30 +352,30 @@ export default class PathsCache extends EventEmitter {
   /**
    * Returns a list of ignore patterns for a directory
    * @param  {String} directoryPath
-   * @return {String[]}
+   * @return {Promise<string[]>}
    * @private
    */
-  _getIgnorePatterns(directoryPath) {
+  async _getIgnorePatterns(directoryPath) {
     const patterns = []
 
     if (this.config.shouldIgnoredNames) {
-      this.config.ignoredNames.forEach((pattern) => patterns.push(pattern))
+      patterns.push(...this.config.ignoredNames)
     }
 
     if (this.config.excludeVcsIgnoredPaths) {
       try {
-        const gitIgnore = fs.readFileSync(directoryPath + "/.gitignore", "utf-8")
-        gitIgnoreParser(gitIgnore).forEach((pattern) => patterns.push(pattern))
+        const gitIgnoreContent = await readFile(join(directoryPath, ".gitignore"), "utf-8")
+        patterns.push(...gitIgnoreParser(gitIgnoreContent))
```

Comment:
This now works. But I need to review the specs of `.gitignore` to understand if `folder` is the given in `.gitignore` it means `**/folder/**` or `absolute_folder/**`

Reply:
I find a project that does this! 🎉 

https://github.com/EE/gitignore-to-glob